reconcile/utils/terrascript_aws_client.py

```diff
@@ -5321,16 +5321,27 @@ def populate_tf_resource_rosa_authenticator(self, spec):
         )
         tf_resources.append(cognito_resource_server_gateway_resource)
 
-        # POOL CLIENT
-        cognito_user_pool_client = aws_cognito_user_pool_client(
+        # OCM POOL CLIENT
+        ocm_cognito_user_pool_client = aws_cognito_user_pool_client(
             "userpool_client",
             name=f"ocm-{identifier}-pool-client",
             user_pool_id=f"${{{cognito_user_pool_resource.id}}}",
             callback_urls=[f"{bucket_url}/token.html"],
             depends_on=["aws_cognito_resource_server.userpool_gateway_resource_server"],
             **pool_client_args,
         )
-        tf_resources.append(cognito_user_pool_client)
+        tf_resources.append(ocm_cognito_user_pool_client)
+
+        # INSIGHTS POOL CLIENT
+        insights_cognito_user_pool_client = aws_cognito_user_pool_client(
+            "ins_userpool_client",
+            name=f"insights-{identifier}-pool-client",
+            user_pool_id=f"${{{cognito_user_pool_resource.id}}}",
+            callback_urls=[f"{bucket_url}/token.html"],
```

Comment:
We'll have two `callback_urls` for Insights here, per environment. These will likely need to be parameterized different from the `bucket_url`, as our redirect uri won't be an s3 bucket asset directly, but rather our domain, and API Gateway will handle the rest.

We can supply these to you, but I'm assuming there will need to be some changes to how they're consumed?

cc: @lphiri

Reply:
>I'm assuming there will need to be some changes to how they're consumed?

Probably. From what you're describing, there's probably going to have to be a new "rosa-authenticator" input called "insights_callback_urls".

In production, for example, that new property would be defined here: https://gitlab.redhat-osd.local/app-sre/app-interface/-/blob/master/data/services/ocm/namespaces/uhc-production.yml#L182-193

Do you know what those URLs would be stage and prod for you guys?

Reply:
What I posted here is probably null and void after having read: https://github.com/app-sre/qontract-reconcile/pull/3196/files#r1107992992